hudi-spark-datasource/hudi-spark3.3.x/src/main/scala/org/apache/spark/sql/parser/HoodieSpark3_3ExtendedSqlAstBuilder.scala

```diff
@@ -3327,6 +3327,145 @@ class HoodieSpark3_3ExtendedSqlAstBuilder(conf: SQLConf, delegate: ParserInterfa
       position = Option(ctx.colPosition).map(pos =>
         UnresolvedFieldPosition(typedVisit[ColumnPosition](pos))))
   }
+
+  /**
```

Comment:
I assume the SQL parsing of INDEX SQL statement should not be different across Spark versions.

Reply:
Yea, it should be same